src/nfa/nfa.cc

```diff
@@ -555,3 +555,14 @@ bool Nfa::is_identical(const Nfa& aut) const {
     }
     return delta == aut.delta;
 }
+
+Nfa& Nfa::complement_deterministic(const OrdVector<Symbol>& symbols, std::optional<State> sink_state) {
+    const size_t num_of_states{ this->num_of_states() };
+    if (initial.empty()) { // The automaton has no reachable states (accepting an empty language).
+        // Insert a single initial sink state.
+        initial.insert(num_of_states);
+    }
```

Comment:
Should this happen even if `sink_state` is defined?

Reply:
It should. It is not important here whether the new state will be the sink state to be used or not (since we use `std::optional::value_or()`, we use the defined sink state, or a new state equal to `num_of_states()`), but rather that we have at least one initial state. Otherwise, the complement (from the point of view of regular languages) would not be computed correctly: it would not accept the empty string, nor the words over the alphabet.

However, `completeness` is a property of the automaton representing the language rather than the language itself. So it should make sense to create an arbitrary graph without any initial states, make it complete and only then add initial and final states depending on some condition, for example. But this does not apply to complement. To ensure the complement is computed correctly, but the `make_complete()` can be computed on automata without initial states, we need to add an initial state.  After that, making the automaton complete and swapping initial and final states will make the automaton accept the empty string and optionally words over the alphabet.

Reply:
I am just thinking whether the defined `sink_state` should not become initial state, instead of creating a new one.

Reply:
Good call. Yeah, we can spare one more state if we do this. Fixed.